.github/workflows/build-cmake.yaml

```diff
@@ -0,0 +1,208 @@
+name: Build CMake
+
+on:
+  workflow_dispatch: # Manual trigger
+  push:
+    branches:
+      - master
+  pull_request:
+    paths-ignore:
+      - .gitignore
+      - .github/**
+      - '!.github/workflows/build-cmake.yaml'  # Run check on self change
+      - CONTRIBUTORS
+      - LICENSE
+      - NOTICE
+      - README.md
+      - android/**
+      - iphone/**
+      - data/strings/**
+      - docs/**
+      - packaging/**
+      - pyhelpers/**
+      - tools/**
+      - '!tools/python/test_server/**'
+      - xcode/**
+
+# Cancels previous jobs if the same branch or PR was updated again.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        cmake-build-type: [ Debug, RelWithDebInfo ]
+        environment:
+          - os: ubuntu-24.04
+            cc: clang-18
+            cxx: clang++-18
+            disable-unity-build: ON
+            run-coverage: OFF
+
+          - os: ubuntu-24.04
+            cc: clang-18
+            cxx: clang++-18
+            disable-unity-build: OFF
+            run-coverage: OFF
+
+          - os: ubuntu-24.04
+            cc: clang-18
+            cxx: clang++-18
+            disable-unity-build: OFF
+            run-coverage: ON
```

Comment:
No need to run the same job without coverage.

Running both release and debug with coverage is not needed. 

Coverage adds +3 minutes for ci, maybe it is better to disable it until someone will need it.